styx-scheduler-service/src/main/java/com/spotify/styx/BackfillTriggerManager.java

```diff
@@ -223,11 +226,45 @@ boolean triggerNextPartitionAndProgress(StorageTransaction tx,
     return true;
   }
 
+  private boolean isAllTriggered(Backfill backfill, Instant nextTrigger) {
+    return nextTrigger.equals(exclusiveEndTrigger(backfill));
+  }
+
+  private Instant exclusiveEndTrigger(Backfill backfill) {
+    final Instant exclusiveLastTrigger;
+    if (backfill.reverse()) {
+      exclusiveLastTrigger = previousInstant(backfill.start(), backfill.schedule());
+    } else {
+      exclusiveLastTrigger = backfill.end();
+    }
+    return exclusiveLastTrigger;
+  }
+
   private void storeBackfill(Backfill backfill) {
     try {
       storage.storeBackfill(backfill);
     } catch (IOException e) {
       LOG.warn("Failed to store updated backfill {}", backfill.id(), e);
     }
   }
+
+  private static Instant getNextPartition(Backfill momentBackfill,
+                                   Instant momentNextTrigger,
+                                   boolean reversed) {
+    return reversed
+        ? previousInstant(momentNextTrigger, momentBackfill.schedule())
+        : nextInstant(momentNextTrigger, momentBackfill.schedule());
+  }
+
+  private static boolean capacityReached(Backfill momentBackfill, Instant initialNextTrigger,
```

Comment:
As far as code-style goes, I don't mind using either ternary of if-else, but let's stick to one and the same, where it applies. `exclusiveEndTrigger`, `getNextPartition` and `capacityReached` can all be written using the same notation